mealprep/server.js

```diff
@@ -8,44 +8,97 @@ const app = express();
 const mysql = require('mysql')
 
 const connection = mysql.createConnection({
-  host: process.env.DATABASE_HOST,
-  user: process.env.DATABASE_USER,
-  password: process.env.DATABASE_PASSWORD
+    host: process.env.DATABASE_HOST,
+    user: process.env.DATABASE_USER,
+    password: process.env.DATABASE_PASSWORD
 })
 
 connection.connect((err) => {
-  if (err) throw err;
-  console.log('Connected!');
+    if (err) throw err;
+    console.log('Connected!');
 });
 
 app.use(cors());
 
 app.post('/login', (req, res) => {
-  var result = false;
-  var getDBInfo = function(callback) {
-
-  		let sql = 'SELECT * FROM gainsday.User WHERE Username = ' + String(req.query.username) + ' AND Password = ' + String(req.query.password);
-	  	connection.query(sql, (err, resp) => {	
-		    if (err) {
-		      console.log("error: ", err);
-		      return callback(err);
-		    }
-
-		    console.log(sql)
-
-		    if (resp.length) {
-		      console.log("found user: ", resp[0]);
-		      result = true;
-		    }
-
-		    callback(null, result);
-	  	});
-	}
-
-	getDBInfo(function (err, result) {
-		console.log({success: result})
-		res.send({success: result});
-	});
+    var result = false;
+    var maxUserID;
+
+    var getDBInfo = function(callback) {
+
+        let sql = 'SELECT * FROM gainsday.User WHERE Username = ' + String(req.query.username) + ' AND Password = ' + String(req.query.password);
+        connection.query(sql, (err, resp) => {
+            if (err) {
+                console.log("error: ", err);
+                return callback(err);
+            }
+
+            console.log(sql)
+
+            if (resp.length) {
+                console.log("found user: ", resp[0]);
+                result = true;
+            }
+
+            callback(null, result);
+        });
+    }
+
+    getDBInfo(function(err, result) {
+        console.log({ success: result })
+        res.send({ success: result });
+    });
+});
+
+app.post('/uploadNewUser', (req, res) => {
+    var result = false;
+    var getDBInfo = function(callback) {
+
+        let sql = 'SELECT MAX(User_id) FROM gainsday.User';
+        connection.query(sql, (err, resp) => {
+            if (err) {
+                console.log("error: ", err);
+                return callback(err);
+            }
+
+            console.log(sql)
+
+            if (resp.length) {
+                console.log("Previous highest user id: ", resp);
+                result = resp;
+            }
+
+            callback(null, result);
+        });
+    }
+
+    var postDBInfo = function(callback) {
+
+        let sql = "INSERT INTO gainsday.User (User_id, Username, Password, Email) VALUES (" + maxUserID + "," + String(req.query.username) + ", " + String(req.query.password) + ", " + String(req.query.email) + ");";
```

Comment:
"INSERT INTO gainsday.User (Username, Password, Email) VALUES (" + String(req.query.username) + ", " + String(req.query.password) + ", " + String(req.query.email) + ");";

if you leave out the user_id field it should automatically increment

Reply:
the line above should work^